lib/request-wrapper.ts

```diff
@@ -36,10 +37,29 @@ import {
 import logger from './logger';
 import { streamToPromise } from './stream-to-promise';
 
+/**
+ * Retry configuration options.
+ */
+export interface RetryOptions {
+  /**
+   * Maximum retries to attempt.
+   */
+  maxRetries?: number;
+
+  /**
+   * Ceiling for the retry delay (in ms) - delay will not exceed this value.
+   */
+  maxRetryInterval?: number;
+}
+
 export class RequestWrapper {
-  private axiosInstance;
+  private axiosInstance: AxiosInstance;
 
-  private compressRequestData: boolean;
+  private retryInterceptorId: number;
+
+  private retryConfig: rax.RetryConfig;
```

Comment:
I wonder if we should rename this field to be "raxConfig" to make it really clear that it is the config used with retry-axios (rax).

Reply:
I think that's a good idea!